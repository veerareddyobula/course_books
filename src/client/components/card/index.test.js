import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import BookDisplayCard from './index';

const mockStore = configureStore([]);

describe('BookDisplayCard Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({});
    const details = {
      titles: 'Letters from a Self-Made Merchant to His Son',
      queries: 'life sentences book',
      book_id: 2,
      author: 'Anna Quindlen',
      id: 2,
      summary: 'The Book in Three Sentences: The only thing you have that nobody else has is control of your life. The hardest thing of all is to learn to love the journey, not the destination. Get a real life rather than frantically chasing the next level of success.'
    };
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <BookDisplayCard details={details} />
      </Provider>
    );
  });
  it('BookDisplayCard Component Snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
