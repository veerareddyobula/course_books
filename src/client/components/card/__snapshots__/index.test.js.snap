// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDisplayCard Component BookDisplayCard Component Snapshot 1`] = `
<div
  className="card"
>
  <div>
    Title
  </div>
  <div>
    Letters from a Self-Made Merchant to His Son
  </div>
  <div>
    Author
  </div>
  <div>
    Anna Quindlen
  </div>
  <div>
    Short Summary
  </div>
  <div
    className="truncate"
  >
    The Book in Three Sentences: The only thing you have that nobody else has is control of your life. The hardest thing of all is to learn to love the journey, not the destination. Get a real life rather than frantically chasing the next level of success.
  </div>
</div>
`;
